api/src/main/java/org/pmiops/workbench/api/WorkspacesController.java

```diff
@@ -495,17 +495,24 @@ private FirecloudWorkspaceId generateFirecloudWorkspaceId(String namespace, Stri
 
   @Override
   public ResponseEntity<WorkspaceResponseListResponse> getWorkspaces() {
-    // TODO: use FireCloud to determine what workspaces to return, instead of just returning
-    // workspaces from our database.
+    org.pmiops.workbench.db.model.Workspace dbWorkspace;
+    List<org.pmiops.workbench.firecloud.model.Workspace> fcWorkspaces;
     User user = userProvider.get();
     List<WorkspaceResponse> responseList = new ArrayList<WorkspaceResponse>();
     if (user != null) {
-      for (WorkspaceUserRole userRole : user.getWorkspaceUserRoles()) {
-        // TODO: Use FireCloud to determine access roles, not our DB
+      try {
+        fcWorkspaces = firecloudService.getWorkspaces();
+      } catch (org.pmiops.workbench.firecloud.ApiException e) {
+        throw ExceptionUtils.convertFirecloudException(e);
+      }
+      for ( org.pmiops.workbench.firecloud.model.Workspace fcWorkspace : fcWorkspaces) {
         WorkspaceResponse currentWorkspace = new WorkspaceResponse();
-        currentWorkspace.setWorkspace(TO_CLIENT_WORKSPACE.apply(userRole.getWorkspace()));
-        currentWorkspace.setAccessLevel(userRole.getRole());
-        responseList.add(currentWorkspace);
+        dbWorkspace = workspaceService.get(fcWorkspace.getWorkspaceNamespace(), fcWorkspace.getWorkspaceName());
```

Comment:
We can filter on namespace so that (at least for now) we only look in the database for workspaces with the namespace for the free tier billing project for the user. That way we'll skip doing lookups for public workspaces / other workspaces returned.

In future we may need to change this logic to filter on some tag we set on workspaces, or a list of namespaces for AofU projects for the user.

Reply:
I think that's also going to have the problem of only pulling user created workspaces, and not workspaces shared to them.

Reply:
+1 to the idea of tagging workspaces. It's not foolproof (users in FC could use the same tag inside FC), but does give us a way to query for a narrower workspaces and possibly mitigating the fear of pulling back N workspaces every time this method is called.

Reply:
OK. @UrsaStutsman, if we took the approach we described for saving UUIDs to workspaces, we could also use it to set tags on existing FC workspaces, that could subsequently be used for filtering purposes.

One annoying thing: it doesn't look like you can set a tag on a workspace when you create it? If so it's possible we could create a workspace but fail to tag it. Wondering if we should use attributes instead?